### PR TITLE
Github Actions: Generate documentation on release

### PR DESCRIPTION
**Pull request by adam-fowler**
Use a github action to generate documentation whenever a new release is published.

Made a couple of changes to the process while doing this.
- Split `gen-doc.sh` into two files `build-doc.sh` and `commit-doc.sh`
- Github action runs `build-doc.sh`
- Make use of latest jazzy which allows multiple sourcekitten files as input
- Include some AWSSDKSwiftCore documentation in the docs.

### CODE REVIEW

**Comment by adam-fowler** [comment]
We have one problem with this PR at the moment. The process takes more than an hour and the GITHUB_TOKEN expires after one hour

**Comment by adam-fowler** [comment]
Sorted the issue with the process taking too long by running the sourcekitten json generation across 8 instances

File: .github/workflows/gen-docs.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
+name: Generate Documentation
+
+on:
+  release:
+    types: [published]
+    branches: "master"
```
Comment (fabianfett) [inline .github/workflows/gen-docs.yml]: We do release from other branches, don't we?

File: .github/workflows/gen-docs.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+      with:
+        fetch-depth: 1
+    - name: Install
+      run: |
+        gem install jazzy
+        npm install lodash fs globby
+    - name: Chunk 1
```
Comment (fabianfett) [inline .github/workflows/gen-docs.yml]: Maybe name should be zero based as well, would make reading easier.

File: .github/workflows/gen-docs.yml (hunk 1 above)
Comment (adam-fowler) [reply .github/workflows/gen-docs.yml]: We do, but I wanted to get master working first.

File: .github/workflows/gen-docs.yml (hunk 2 above)
Comment (adam-fowler) [reply .github/workflows/gen-docs.yml]: not sure it makes a huge difference

Comment (adam-fowler) [reply .github/workflows/gen-docs.yml]: > How long would the gen-docs process take if it wouldn't be parallelized?

It takes 75-80 mins which is too long as the GITHUB_TOKEN expires after an hour. Unfortunately when we split across multiple instances we have to re-build the whole of aws-sdk-swift on each instance.

Comment (fabianfett) [reply .github/workflows/gen-docs.yml]: > It takes 75-80 mins which is too long as the GITHUB_TOKEN expires after an hour. Unfortunately when we split across multiple instances we have to re-build the whole of aws-sdk-swift on each instance.

Maybe we should add this as a comment in the yaml file will make it much clearer for future us.

**Comment by adam-fowler** [comment]
Yeah makes sense

File: .github/workflows/gen-docs.yml (hunk 2 above)
Comment (adam-fowler) [reply .github/workflows/gen-docs.yml]: done

**Comment by Yasumoto** [comment]
This is awesome 👏

**Pull request merged**